### PR TITLE
Add Github Actions workflows and composite actions for packs to use

### PR DESCRIPTION
**Pull request by cognifloyd**
This is part of migrating from CircleCI to Github Actions: https://github.com/StackStorm/community/issues/63

I just pushed a commit that switches loading commits from @gha to @master. So, workflows in other repos won't work until this is merged.

`.github/actions/py-dependencies/requirements*.txt` are copies of `.circle/requirements*.txt` with the expectation that `.circle/` will eventually be removed. This keeps the dependencies close to where they are used which simplifies accessing them and maintaining them.

### CODE REVIEW

File: .github/workflows/index-update.yaml [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+    - name: Checkout exchange-tools repo
+      uses: actions/checkout@v2
+      with:
+        # TODO: switch to StackStorm-Exchange once these are merged:
+        # https://github.com/StackStorm-Exchange/exchange-tools/pull/3
+        # https://github.com/StackStorm-Exchange/exchange-tools/pull/2
+        #repository: StackStorm-Exchange/exchange-tools
+        repository: cognifloyd/exchange-tools
+        ref: ${{ inputs.exchange_tools_branch }}
```
Comment (cognifloyd) [inline .github/workflows/index-update.yaml:98]: We need to merge these before we merge this PR.

**Comment by cognifloyd** [comment]
We did not touch the sections of code that pylint is complaining about, so I'm not fixing the warnings identified by pylint in the CircleCI run.

**Review by mweinberg-cm — APPROVED** [review]

**Comment by mweinberg-cm** [comment]
@cognifloyd @lm-ydubler and me just had the call to review the PRs and agreed that the code looks good and is ready to be merged. Great job by both of you guys!

**Comment by winem** [comment]
@cognifloyd @lm-ydubler and me just had the call to review the PRs and agreed that the code looks good and is ready to be merged. Great job by both of you guys!

**Review by winem — APPROVED** [review]

**Pull request merged**